src/components/UI/Modal/Modal.jsx

```diff
@@ -0,0 +1,45 @@
+import styles from "./Modal.module.css";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faClose } from "@fortawesome/free-solid-svg-icons";
+import { useModal } from "../../../context/modal-context";
+import { useEffect, useState } from "react";
+
+export const Modal = (props) => {
+  const { modalStateDispatch, modalState } = useModal();
+  const [isMounted, setIsMounted] = useState(false);
+  useEffect(() => {
+    setIsMounted(true);
+  }, []);
+  return (
+    <>
+      <div className={`${styles["modal"]} ${isMounted && styles["show"]}`}>
+        <div
+          className={styles["modal-close-btn"]}
+          onClick={() => {
+            setIsMounted(false);
+            setTimeout(() => {
+              modalStateDispatch({type: "CLOSE"});
+            }, 500);
```

Comment:
Suggestion: setTimeout on onClick will lead to creating multiple timers. Instead of it, you can handle this timer in useEffect.